wallet/src/main/java/com/android/identity_credential/wallet/ui/prompt/passphrase/PassphrasePinScreen.kt

```diff
@@ -0,0 +1,364 @@
+    // the mnemonic part of the keypad button (letters under each number)
+    val chars: String? = null,
+    // show an image instead of a number and chars; provide buttonId below to find it in a list
+    val imageResId: Int? = null,
+    // used to identify the button when an imageResId is provided (and number and chars are null)
+    val buttonId: String? = null,
+    // for accessibility usually for image buttons
+    val description: String? = null,
+) {
+    /**
+     * An empty KeyPadButton renders a Spacer instead.
+     */
+    fun isEmpty() = number == null && chars == null && imageResId == null && description == null
+}
+
+/**
+ * Composes a full screen with PassphraseFieldEntry and custom numerical key pad to use for PINs.
+ *
+ * With fixed length constraints the user cannot tap on a "Done" button because it is automatically
+ * submitted when user enters the required number of digits - so only the Cancel button is
+ * composed on the bottom right that changes to "Delete" the moment digits start to be tapped.
+ * Tapping on Delete will delete the last tapped digit until there's no digits remaining/showing and
+ * the button's text changes back to "Cancel" that cancels the prompt.
+ *
+ * For non-fixed constraints the user has the option to submit the entered PIN at any point unless
+ * the maximum number of digits has been entered that automatically submits the PIN. The
+ * Cancel/Delete button is composed on the bottom left and the "Done" button on the bottom right.
+ */
+@Composable
+fun PassphrasePinScreen(
+    title: String,
+    content: String,
+    constraints: PassphraseConstraints,
+    onSubmitPin: (String) -> Unit,
+    onCancel: () -> Unit
+) {
+    if (!constraints.requireNumerical) {
+        throw IllegalStateException("Passphrase Pin Screen is for numerical passphrases only.")
+    }
+
+    val currentPin = remember { mutableStateOf("") }
+    val showDelete = remember { mutableStateOf(false) }
+
+    /**
+     * Called for updating the text of the Cancel/Delete button to show "Delete" text or not when
+     * user taps on a KeyPad button or deletes a digit.
+     */
+    fun updateShowDelete() {
+        showDelete.value = currentPin.value.isNotEmpty()
+    }
+
+    /**
+     * Tapping on "Delete" button sets the [currentPin] length to be 1 less character.
+     */
+    fun onBackspace() {
+        currentPin.value = currentPin.value.substring(0, currentPin.value.lastIndex)
+    }
+
+    Column {
+        Spacer(modifier = Modifier.height(50.dp))
+        PassphrasePinScreenHeader(title = title, content = content)
+
+        Spacer(modifier = Modifier.height(50.dp))
+        // prevent the keyboard from showing up in PassphraseEntryField and set the text to show
+        PassphraseEntryField(
+            constraints = constraints,
+            disableKeyboard = true,
+            setText = currentPin.value,
+            onChanged = { _, _, _ -> /* will never get called */ },
+        )
```

Comment:
I think the code would be easier to read if you just used `BasicTextField` here ... that way you also don't have to add `disableKeyboard` or `setText` to `PassphraseEntryField`.